partial_helper.go

```diff
@@ -0,0 +1,49 @@
+package plush
+
+import (
+	"html/template"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// PartialFeeder is callback function should implemented on application side.
+type PartialFeeder func(string) (string, error)
+
+func partialHelper(name string, data map[string]interface{}, help HelperContext) (template.HTML, error) {
+	if help.Context == nil || help.Context.data == nil {
+		return "", errors.New("invalid context. abort")
+	}
+	for k, v := range data {
+		help.Context.data[k] = v
+	}
+
+	pf, ok := help.Context.data["partialFeeder"].(func(string) (string, error))
```

Comment:
Type check against PartialFeeder here

Reply:
Can I just use below here?

```
var pf PartialFeeder
```

I think it is just `func(string) (string, error)` so if the interface `data["partialFeeder"]` not match, it will not `ok` anyway.